spring-integration-core/src/main/java/org/springframework/integration/dispatcher/PartitionedDispatcher.java

```diff
@@ -164,7 +164,7 @@ public void shutdown() {
 	}
 
 	@Override
-	@SuppressWarnings("NullAway")
+	@SuppressWarnings("NullAway") // Dataflow analysis limitation!
```

Comment:
Yeah... That's not what I meant, especially not with a `!` 😉 
How about this?
```suggestion
	@SuppressWarnings("NullAway") // The partitions map never returns null according to partition hash
```

Reply:
Thank you for the clarification.   I thought that message was a bit terse.   will update.